crypto/bls12381_utils.c

```diff
@@ -568,6 +568,73 @@ void ep2_sum_vector(ep2_t jointy, ep2_st* y, const int len){
     }
 }
 
+// Verifies the validity of 2 SPoCK proofs and 2 public keys.
+// Membership check in G1 of both proofs is verified in this function.
+// Membership check in G2 of both keys is not verified in this function.
+// the membership check in G2 is separated to allow optimizing multiple verifications 
+// using the same public keys.
+int bls_spock_verify(const ep2_t pk1, const byte* sig1, const ep2_t pk2, const byte* sig2) {  
+    ep_t elemsG1[2];
+    ep2_t elemsG2[2];
+
+    // elemsG1[0] = s1
+    ep_new(elemsG1[0]);
+    if (ep_read_bin_compact(elemsG1[0], sig1, SIGNATURE_LEN) != RLC_OK) 
+        return INVALID;
+
+    // check s1 is on curve and in G1
+    if (check_membership_G1(elemsG1[0]) != VALID) // only enabled if MEMBERSHIP_CHECK==1
+        return INVALID;
+
+    // elemsG1[0] = s2
+    ep_new(elemsG1[1]);
+    if (ep_read_bin_compact(elemsG1[1], sig2, SIGNATURE_LEN) != RLC_OK) 
+        return INVALID;
+
+    // check s2 is on curve and in G1
+    if (check_membership_G1(elemsG1[1]) != VALID) // only enabled if MEMBERSHIP_CHECK==1
+        return INVALID; 
+
+    // elemsG2[1] = pk1
+    ep2_new(elemsG2[1]);
+    ep2_copy(elemsG2[1], (ep2_st*)pk1);
+
+    // elemsG2[0] = pk2
+    ep2_new(&elemsG2[0]);
```

Comment:
here, you use the memory address `&elemsG2[0]` of point pk2 while above, you use the value:
https://github.com/onflow/flow-go/blob/de7518dbbeffc54db495dc842c73181ea3db0506/crypto/bls12381_utils.c#L599

is that intentional?

Reply:
Good catch! It doesn't matter with the Relic settings we're using now but it does matter if I change the settings.